jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/ExceptionUtil.java

```diff
@@ -76,6 +77,68 @@ public static <T extends Throwable> T as(Class<T> type, Throwable throwable) thr
         }
     }
 
+    /**
+     * Create a copy of a Throwable, keeping the cause but discarding suppressed exceptions.
+     * The copy will be of the same type if it has a constructor that takes a String
+     * message, otherwise it will be of the same supertype (e.g. IOException).
+     * @param failure The Throwable to copy
+     * @return A copy of the Throwable with same cause, but no suppressed exceptions.
+     */
+    public static Throwable copyOf(Throwable failure)
+    {
+        if (failure == null)
+            return null;
+
+        Throwable cause = failure.getCause();;
+
+        Class<? extends Throwable> failureType = failure.getClass();
+        while (failureType != Exception.class && failureType != RuntimeException.class && failureType != Error.class && failureType != Throwable.class)
```

Comment:
We can do these 2 pages of code with 2 lines of calling the JDK methods.
I don't see the reason to not call the JDK methods?
We should just call `failureType.getConstructor(String.class)` and then `initCause()`.
Am I missing something?